src/helpers/workspace.ts

```diff
@@ -17,6 +25,42 @@ export function getWorkspaceRoot(ignoreException: boolean = false): string | und
   return workspaceRoot;
 }
 
+export async function getWorkspaces(): Promise<TruffleWorkspace[]> {
+  const workspaces: TruffleWorkspace[] = [];
+
+  await Promise.all(
+    workspace.workspaceFolders!.map(async (ws) => {
+      workspaces.push(...(await getWorkspaceFiles(ws.uri.fsPath)));
+    })
+  );
+
+  if (workspaces.length === 0) {
+    const error = new Error(Constants.errorMessageStrings.VariableShouldBeDefined("Workspace root"));
+    Telemetry.sendException(error);
+    throw error;
+  }
+
+  return workspaces;
+}
+
 export function isWorkspaceOpen(): boolean {
   return !!(workspace.workspaceFolders && workspace.workspaceFolders[0].uri.fsPath);
 }
+
+async function getWorkspaceFiles(dirPath: string): Promise<TruffleWorkspace[]> {
+  const truffleWorkSpaces: TruffleWorkspace[] = [];
+
+  const files = glob.sync(`${dirPath}/**/${Constants.defaultTruffleConfigFileName}`, {
+    ignore: Constants.workspaceIgnoredFolders,
+  });
+
+  files.forEach((file) => {
```

Comment:
you can use a `.map<Type>(...` here instead and it will do the same thing without having to use a push and const declaration on line 51.

Reply:
```
const truffleWorkSpaces: TruffleWorkspace[] =
 files.map((f) => (return {
      dirName: path.dirname(file).split(path.sep).pop()!.toString(),
      workspace: Uri.parse(path.dirname(file)),
      truffleConfig: Uri.parse(file),
    });

```

Reply:
Thank you @michaeljohnbennett! :)

Reply:
I will do this way before merge this branch :)